### ISSUE TITLE
`Get Variable Value` fails if variable name contains another variable and resolving it fails

### ISSUE THREAD

**Issue by haryw4** [robotframework/robotframework#4175]
With RFW 3 this was working fine when inner dictionary entry did not exist : ${var[${dict['A']}]} -> default returned
With RFW 4 this fails with : Resolving variable '${dict['A']}' failed: KeyError: 'A'

This contradicts the purpose of this keyword. It should return given default whenever it is not possible to resolve the given term.

**Comment by peerasut** [comment]
I have RF4.0 on python 3.10, it works fine. None is returned.
<img width="555" alt="image" src="https://user-images.githubusercontent.com/68002612/165774166-56468fef-d70e-4108-9592-b549f234435e.png">

If you still need help, could you share anything about your code or python version you're using?

**Comment by pekkaklarck** [comment]
The problem occurs with nested variables. For example this fails:
```robotframework
*** Test Cases ***
Example
    ${d} =    Create Dictionary    a=1    b=2
    Get Variable Value    ${d['${d['x']}']}
```
I don't consider this too severe because the problem is that a key is missing from a dictionary, not that the keyword wouldn't work with a valid variable. That said, it's pretty inconsistent that missing keys don't cause problems without nested variables (e.g. `${d['x']}` works fine). I validated that the above works with RF 3.2.2 so this is a regression as well. I'll see could this be fixed in RF 5.0.1.

A workaround to this issue is first just checking does the variable containing a dictionary exist. If it does, then dictionary related keywords can be used for getting values out from it.
